src/sql/sql_provider_datafusion/mod.rs

```diff
@@ -276,24 +299,35 @@ impl<T, P> SqlExec<T, P> {
             let filter_expr = self
                 .filters
                 .iter()
-                .map(|f| match f {
-                    // DataFusion optimization uses aliases to preserve original expression names during optimizations and type coercions:
-                    // https://github.com/apache/datafusion/issues/3794
-                    // If there is a filter with additional alias information, we must use the expression only
-                    // as original expression name is unnecessary and the alias can't be part of the WHERE clause
-                    Expr::Alias(alias) => expr::to_sql_with_engine(&alias.expr, self.engine),
-                    _ => expr::to_sql_with_engine(f, self.engine),
+                .map(|f| {
+                    Unparser::new(self.dialect.as_ref())
+                        .expr_to_sql(f)
+                        .map(|e| e.to_string())
                 })
-                .collect::<expr::Result<Vec<_>>>()
-                .context(UnableToGenerateSQLSnafu)?;
-            format!("WHERE {}", filter_expr.join(" AND "))
+                .collect::<DataFusionResult<Vec<String>>>()
+                .context(UnableToGenerateSQLSnafu)?
+                .join(" AND ");
+            format!("WHERE {}", filter_expr)
         };
 
         Ok(format!(
             "SELECT {columns} FROM {table_reference} {where_expr} {limit_expr}",
-            table_reference = self.table_reference.to_quoted_string()
+            table_reference = self.table_name_escaped(),
         ))
     }
+
+    fn table_name_escaped(&self) -> String {
+        self.table_reference.to_quoted_string()
+    }
+
+    fn column_name_escaped(&self, column_name: &str) -> String {
+        match self.engine {
+            Some(Engine::ODBC) => column_name.to_string(),
+            _ => {
+                format!("\"{}\"", column_name)
+            }
```

Comment:
This is probably the only reason we have the Engine now right? I'm pretty sure the Dialect has a method to determine the column escaping character that we could use.